src/test/compile-fail/empty-struct-unit-expr.rs

```diff
@@ -24,10 +24,10 @@ enum E {
 fn main() {
     let e2 = Empty2(); //~ ERROR expected function, found `Empty2`
     let e4 = E::Empty4();
-    //~^ ERROR `E::Empty4` is being called, but it is not a function
+    //~^ ERROR expected a function, but `E::Empty4` is `E`
```

Comment:
@petrochenkov 

> `E::Empty4` cannot be called, because `E` doesn't implement `Fn`

True. Still, it seems to include a lot more jargon. Perhaps with a suggestion it's good enough.